### ISSUE TITLE
Array of calculations

### ISSUE THREAD

**Issue by aiida-bot** [aiidateam/aiida_core#111]
Originally reported by: **Giovanni Pizzi (Bitbucket: [pizzi](https://bitbucket.org/pizzi), GitHub: [giovannipizzi](https://github.com/giovannipizzi))**

----------------------------------------

Due to policies of some clusters, sometimes only few jobs can be submitted, and they require to run with a script similar to the following:

```
mpirun -np 16 exec < input1 >out1 &
mpirun -np 16 exec < input2 >out2 &
mpirun -np 16 exec < input3 >out3 &
wait
```
This can be achieved as follows:

* add a ```only_send=True``` parameter to ```.submit()``` that will send all files of each calculation but will not execute the ```qsub``` command.
* Set the state in a new SENTONLY state
* Create a new calculation of type, e.g., ArrayCalculation, having as input the RemoteData of the parent jobs, in SENTONLY state. The script will run each of the parent scripts with ```&``` at the end, end then add ```wait```
* A simple parser of the ArrayCalculation will simply switch the parent calculation states from SENTONLY to COMPUTED, so the daemon will then parse them.

Issues:

* one can call directly the scripts of the parent calculation, maybe, but there may be issues with the environment variables (e.g. if different set of modules are loaded, or for the PBS_O_WORKDIR, etc.). Understand if it is possible to load an independent environment for each subjob, and if this is ok.
* If the parent is the RemoteData node, one can create the ArrayCalculation only after submission, but at the moment this is not detectable
* If a child calculation has to be run, it has to be attached to each calculation and not to the ArrayCalculation, since its parser will just trigger the parsing of the parents, but not actually parse them.

----------------------------------------
- Bitbucket: https://bitbucket.org/aiida_team/aiida_core/issue/111

**Comment by DropD** [comment]
This can be achieved using 

```
from aiida.common.datastructures import code_run_modes
calcinfo = CalcInfo()
...

codeinfoA = CodeInfo()
...
codeinfoB = CodeInfo()
...

calcinfo.codes_info = [codeinfoA, codeinfoB, ...]
calcinfo.codes_run_mode = code_run_modes.PARALLEL  # or .SERIAL
```

The documentation for this is hidden in http://aiida-core.readthedocs.io/en/stable/developer_guide/devel_tutorial/code_plugin_qe.html#step-3-prepare-a-text-input. There is a new issue #1000 about improving the documentation for this.

**Comment by DropD** [comment]
Already implemented, therefore closing.

**Comment by broeder-j** [comment]
I'm must say I am not so sure if this is so, because a CODE-PLUGIN (1) and JOB ARRAYS (2) are two different things. Or maybe I miss something here, or from the documentation. 

1. is for if a code nodes several executables, pre or post processing, or has features to execute the same executable in several subdirs or so. (this is implemented)

2. Whereas the job array, and what Giovanni mentions is something you have to decide from the outside and the DAEMON groups several jobs into ONE job script. So far I do not see this 'outside' calculation plugin part. I.E some variable you give to submit, or something else.
